### PR TITLE
dumb_stringstream initialization bugfix.

### PR DESCRIPTION
**Pull request by dvereb**
Solves issues with `dumb_stringstream` in #124.

### CODE REVIEW

**Comment by dvereb** [comment]
I have no idea why this works, just that it works for me.  If someone else could explain I would be so happy! :)

File: src/strconv.cxx [hunk 1]
```diff
@@ -232,8 +232,7 @@ template<typename T> inline void from_string_float(const char Str[], T &Obj)
       S.seekg(0);
       S.clear();
       S.str(Str);
-      S >> result;
-      ok = S.good() || S.eof();
+      ok = static_cast<bool>(S >> result);
```
Comment (egorpugin) [inline src/strconv.cxx]: Better `ok = S;` or `ok = !!S;` or like yours `ok = static_cast<bool>(S);`.

File: src/strconv.cxx [hunk 2]
```diff
@@ -178,7 +178,7 @@ bool valid_infinity_string(const char str[]) noexcept
 template<typename T> class dumb_stringstream : public std::stringstream
 {
 public:
-  dumb_stringstream() : std::stringstream{}
+  dumb_stringstream() : std::stringstream()
```
Comment (egorpugin) [inline src/strconv.cxx]: Today {} sometimes might be a little different thing than ().
In constructor preferred way is (), as I understand.

**Comment by dvereb** [comment]
No, thank you for all that you do. My company relies on this library and I'm glad I could contribute. This is my second ever pull request (and first accepted). It wasn't much, but it felt good to give back! ;)

**Pull request merged**